### ISSUE TITLE
feat: GitHubReleaseVerificationRule - detect action tags not backed by an official GitHub Release

### ISSUE THREAD

**Issue by ultra-supara** [ultra-supara/sisakulint#401]
## 概要

アクション参照のタグが GitHub Releases として正式に存在するかを検証する新ルール。

タグはwrite権限があれば自由に作成・削除できるが、GitHub Releases はより管理された成果物であり、偽タグとの区別に使える。

## 攻撃シナリオ

- 攻撃者が `v0.69.4` 偽タグを作成（GitHub Release は存在しない）
- `v0.70.0` タグを削除（Release が存在していた場合は乖離が生じる）

## 検出ロジック

```
aquasecurity/trivy-action@v0.69.4
  ↓ GitHub Releases API: GetReleaseByTag("v0.69.4")
  → 404 Not Found (Releaseが存在しない)
  → WARNING: tag 'v0.69.4' of 'aquasecurity/trivy-action' has no corresponding
             GitHub Release. This may indicate an unofficial or malicious tag.

または

  ↓ Release は存在するが、Release.TagName のコミットSHA と タグのコミットSHA が不一致
  → WARNING: commit SHA mismatch between tag and Release asset
```

## 実装メモ

- GitHub API: `client.Repositories.GetReleaseByTag`
- false positive 対策: 一部のプロジェクトはタグのみでリリース管理する（例: Docker公式イメージ）。設定で除外リスト対応が必要
- `CommitShaRule` でSHAピン留めを推奨しているが、そのSHAが正規Releaseに対応しているかの検証にも使える

## 優先度

TagCommitReachabilityRule（#前のissue）より偽陽性リスクが高いため、P3扱いが妥当。

## 参考

- https://diary.shift-js.info/trivy-compromise/
